### PR TITLE
Add extra compiler options to build Valhalla tests

### LINKED ISSUE #16031: Compilation error of Valhalla Tests
Failure link
------------

Internal Nightly Build: `Test_openjdkValhalla_j9_extended.functional_x86-64_linux_valhalla_Nightly_testList_1/641/`

Optional info
-------------

- intermittent failure: no 

Failure output (captured from console output)
---------------------------------------------
```
22:37:03      [javac] /home/jenkins/workspace/Test_openjdkValhalla_j9_extended.functional_x86-64_linux_valhalla_Nightly_testList_0/aqa-tests/functional/Valhalla/src/org/openj9/test/lworld/ValueTypeArrayTests.java:58: error: primitive classes are not supported
22:37:03      [javac] 	static primitive class PointPV implements SomeIface {
22:37:03      [javac] 	       ^
22:37:03      [javac]   (use -source 19 or higher to enable primitive classes and pass compiler option: -XDenablePrimitiveClasses)
22:37:04      [javac] /home/jenkins/workspace/Test_openjdkValhalla_j9_extended.functional_x86-64_linux_valhalla_Nightly_testList_0/aqa-tests/functional/Valhalla/src/org/openj9/test/lworld/ValueTypeUnsafeTestClasses.java:27: error: primitive classes are not supported
22:37:04      [javac] 	static primitive class ValueTypeInt {
22:37:04      [javac] 	       ^
22:37:04      [javac]   (use -source 19 or higher to enable primitive classes and pass compiler option: -XDenablePrimitiveClasses)
22:37:04      [javac] /home/jenkins/workspace/Test_openjdkValhalla_j9_extended.functional_x86-64_linux_valhalla_Nightly_testList_0/aqa-tests/functional/Valhalla/src/org/openj9/test/lworld/ValueTypeUnsafeTests.java:267: error: primitive classes are not supported
22:37:04      [javac] 		primitive class NeverInitialized {
22:37:04      [javac] 		^
22:37:04      [javac]   (use -source 19 or higher to enable primitive classes and pass compiler option: -XDenablePrimitiveClasses)
22:37:04      [javac] /home/jenkins/workspace/Test_openjdkValhalla_j9_extended.functional_x86-64_linux_valhalla_Nightly_testList_0/aqa-tests/functional/Valhalla/src/org/openj9/test/lworld/ValueTypeUnsafeTests.java:267: error: ';' expected
22:37:04      [javac] 		primitive class NeverInitialized {
22:37:04      [javac] 		         ^
22:37:04      [javac] 4 errors
```

### PR DESCRIPTION
**Pull request by hangshao0**
Fixes #16031

Signed-off-by: Hang Shao <hangshao@ca.ibm.com>

### CODE REVIEW

**Comment by hangshao0** [comment]
I hard code `-source 20` here because `JDK_VERSION` is `Valhalla` in Valhalla builds.

File: test/functional/Valhalla/build.xml [hunk 1]
```diff
@@ -56,6 +56,7 @@
 			<src path="${src}"/>
 			<src path="${transformerListener}" />
 			<compilerarg line='--add-exports java.base/jdk.internal.misc=ALL-UNNAMED --add-exports java.base/jdk.internal.value=ALL-UNNAMED' />
+			<compilerarg line='-source 20 -XDenablePrimitiveClasses' />
```
Comment (tajila) [inline test/functional/Valhalla/build.xml:59]: is `-XD` intentional? usually its `-X` or `-D`

**Review by tajila — APPROVED** [review]

**Comment by tajila** [comment]
Jenkins test sanity,extended xlinuxval jdknext

**Comment by hangshao0** [comment]
OK, the `Class.isIdentity()` issue mentioned in https://github.com/eclipse-openj9/openj9/issues/16027 is auto merged into openj9 branch from openj9-staging branch yesterday. The PR build failed at:

```
16:00:02  /home/jenkins/workspace/Build_JDKnext_x86-64_linux_valhalla_Personal/src/java.base/share/classes/java/util/Objects.java:202: error: cannot find symbol
16:00:02          return obj.getClass().isIdentity() ||  // Before Valhalla all classes are identity classes
16:00:02                               ^
16:00:02    symbol:   method isIdentity()
```

This failure should be fixed by https://github.com/eclipse-openj9/openj9/pull/16041/

**Comment by hangshao0** [comment]
I have added the change here into https://github.com/eclipse-openj9/openj9/pull/16041 so that all changes can be reviewed and tested there. Closing this PR.

**Pull request closed without merging**